kedro/utils.py

```diff
@@ -46,6 +62,17 @@ def _is_project(project_path: Union[str, Path]) -> bool:
 
 
 def _find_kedro_project(current_dir: Path) -> Any:  # pragma: no cover
```

Comment:
```suggestion
def _find_kedro_project(current_dir: Path) -> Optional[bool]:
```

Thanks to your excellent documentation, we should do a better type hint right?

Reply:
Return type here is actually this: `Union[Path, None]`.

I didn't change it in this PR because `mypy` starts complaining at another place:

```
kedro/ipython/__init__.py:173: error: Incompatible types in assignment (expression has type "Path | None", variable has type "Path")  [assignment]
```

So I decided to not resolve it here to focus this PR only on docstrings and thus simplify review.

Reply:
@datajoely ok so I fixed that too, and pushed one commit on top of this branch. We can always revert if you prefer this PR do be `+++` in terms of diff and docstrings-focused only.